server/resp/src/test/java/org/infinispan/JsonCommandsTest.java

```diff
@@ -0,0 +1,190 @@
+package org.infinispan;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.infinispan.test.TestingUtil.k;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.infinispan.server.resp.CustomStringCommands;
+import org.infinispan.server.resp.SingleNodeRespBaseTest;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.Option;
+
+import io.lettuce.core.RedisCommandExecutionException;
+import io.lettuce.core.api.sync.RedisCommands;
+import io.lettuce.core.json.DefaultJsonParser;
+import io.lettuce.core.json.JsonPath;
+import io.lettuce.core.json.JsonValue;
+import io.lettuce.core.json.arguments.JsonSetArgs;
+
+@Test(groups = "functional", testName = "server.resp.JsonCommandsTest")
+public class JsonCommandsTest extends SingleNodeRespBaseTest {
```

Comment:
Just for completeness, would be nice a test verifying the wrongtype behavior.

Reply:
oh, good point. I think we need a new type to distinguish json from strings

Reply:
I would say we should confirm what REDIS is doing. They may allow bulk string type to also be parsed as JSON.

Reply:
yeah, just checked they don't
```
127.0.0.1:6379> set a '{"k":"v"}'
OK
127.0.0.1:6379> json.get a $
(error) Existing key has wrong Redis type
```